src/main/java/reactor/core/publisher/FluxSink.java

```diff
@@ -65,6 +67,14 @@
 	FluxSink<T> serialize();
 
 	/**
+	 * Attaches a {@link LongConsumer} to this {@link FluxSink} that will observe any request
+	 * to this sink.
+	 * @param consumer the consumer to invoke on each request
+	 * @return {@link FluxSink} that observes requests
+	 */
+	FluxSink<T> onRequest(LongConsumer consumer);
```

Comment:
@akarnokd @smaldini Thank you for the review. I have updated the code to use a serialized sink when `onRequest` is invoked and the `onRequest` consumer is now a `BiConsumer` which passes in the (serialized) sink as well. Hopefully that addresses some of the concerns above. `MonoSink` has been updated to be consistent with `FluxSink`. Added more unit tests for `FluxCreate` to test with multiple threads.